### DISCUSSION TITLE
Session flash not working with poll

### DISCUSSION THREAD

**Discussion by haythembenkhlifa** [livewire/livewire#4713] [1. Help] [answered]
Hi There,

Is it possible to keep the session flash message when using the poll?

the issue is that it disappears quickly I want to keep it until the next page reload.

`session()->flash('success', 'New order created.');`

```
<div  class="d-flex flex-column justify-content-center px-1 px-sm-5 py-5" wire:poll.1000ms>
    @if(session('success'))
    <div class="alert alert-success" role="alert">
        {{session('success')}}
    </div>
    @endif
</div>

```

**Comment by silnarm** [comment]
Polling and session flash variables... interesting can o' worms.
"Reflashing" when polling is probably a sensible feature request, always a risk of unintended consequences of course but nothing very nasty immediately comes to mind.
In the meantime, you can do so yourself:
```
<div class="d-flex flex-column justify-content-center px-1 px-sm-5 py-5" wire:poll.1000ms="reflash">
```
```
    public function reflash() { session()->reflash(); }
```

**Comment by haythembenkhlifa** [comment, marked as answer]
Reflash keeps the data for several requests but my goal is to keep it until a full page refresh.

I stop using the session because in my use case it won't help when I'm using poll.

I create a new variable in the component to store the message

`public $message = null;`

` // session()->flash('success', 'New order created.');`
`$this->message = 'New order created.';`

` 
@if($message)
    <div class="alert alert-success" role="alert">
        {{$message}}
    </div>
    @endif
 `

@silnarm  Thank you.

**Reply by silnarm** [reply]
Yeah, flash in particular is always going to be an issue with Livewire, reflash() will actually keep the vars and make them available for one additional request, so the polling in isolation should have worked, but there are other things going on of course (actions, updates for wired form elements) which each create their own AJAX request, so you would have needed to reflash in all your actions, and define a updated() hook and do it there too (for wire:model updates).

What you have done instead is much nicer :-)
